Rubberduck.Core/AutoComplete/SelfClosingPairCompletion/CodeStringPrettifier.cs

```diff
@@ -1,41 +1,61 @@
-﻿using Rubberduck.Common;
+﻿using System;
+using System.Linq;
+using Rubberduck.Common;
 using Rubberduck.VBEditor.SafeComWrappers.Abstract;
-using System;
+using Rubberduck.VBEditor;
 
 namespace Rubberduck.AutoComplete.SelfClosingPairCompletion
 {
     public class CodeStringPrettifier : ICodeStringPrettifier
     {
-        private readonly ICodeModule _module;
-
-        public CodeStringPrettifier(ICodeModule module)
-        {
-            _module = module;
-        }
-
-        public bool IsSpacingUnchanged(CodeString code, CodeString original)
+        public CodeString Prettify(ICodeModule module, CodeString original)
         {
-            using (var pane = _module.CodePane)
+            var originalCode = original.Code.Replace("\r", string.Empty).Split('\n');
+            var originalPosition = original.CaretPosition.StartColumn;
+            var originalNonWhitespaceCharacters = 0;
+            for (var i = 0; i <= originalPosition - 1; i++)
             {
-                using (var window = pane.Window)
+                if (originalCode[original.CaretPosition.StartLine][i] != ' ')
                 {
-                    //window.ScreenUpdating = false;
-                    _module.DeleteLines(code.SnippetPosition);
-                    _module.InsertLines(code.SnippetPosition.StartLine, code.Code);
-                    //window.ScreenUpdating = true;
+                    originalNonWhitespaceCharacters++;
+                }
+            }
 
-                    pane.Selection = code.SnippetPosition.Offset(code.CaretPosition);
+            var indent = originalCode[original.CaretPosition.StartLine].TakeWhile(c => c == ' ').Count();
+            
+            module.DeleteLines(original.SnippetPosition.StartLine, original.SnippetPosition.LineCount);
+            module.InsertLines(original.SnippetPosition.StartLine, string.Join("\r\n", originalCode));
+            var prettifiedCode = module.GetLines(original.SnippetPosition).Replace("\r", string.Empty).Split('\n');
 
-                    var lines = _module.GetLines(code.SnippetPosition);
-                    if (lines.Equals(code.Code, StringComparison.InvariantCultureIgnoreCase))
+            var prettifiedNonWhitespaceCharacters = 0;
+            var prettifiedCaretCharIndex = 0;
+            for (var i = 0; i < prettifiedCode[original.CaretPosition.StartLine].Length; i++)
+            {
+                if (prettifiedCode[original.CaretPosition.StartLine][i] != ' ')
+                {
+                    prettifiedNonWhitespaceCharacters++;
```

Comment:
whitespace is not only `' '`. It also includes `'\t\h'` and a handful of other characters. Why doe we rely on space only?

Reply:
this also applies for keeping track of the codeline's indent. We're completely ignoring tab when determining indent. Is that because the VBE auto-converts tabs to spaces? If yes, that warrants a comment, if no that's a bug :)

Reply:
VBA converts tabs to spaces AFAIK (will confirm). Way to shut the tabs vs spaces debate :smile: